src/rgw/rgw_op.cc

```diff
@@ -5483,7 +5483,8 @@ void RGWCompleteMultipart::execute()
       }
 
```

Comment:
@fangyuxiangGL: Is it possible to compact both checks that lead to "compression type changed error" to single condition?
I envision something along these lines:

      bool part_compressed = obj_part.cs_info.compression_type != "none";            
      if ( obj_iter != obj_parts.begin() && /*first part can't conflict*/
        ( part_compressed != compressed ||
          cs_info.compression_type != obj_part.cs_info.compression_type /*after that can't change*/
        ) ) {
        //failure action here
      }
      if (part_compressed) {
        //compression code here
      }

Reply:
@aclamk 

ok, I will make the codes more concise.

Reply:
@aclamk @cbodley 

addressed